app/assets/javascripts/admin/addon/components/bulk-user-delete-confirmation.gjs

```diff
@@ -0,0 +1,199 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
+import DButton from "discourse/components/d-button";
+import DModal from "discourse/components/d-modal";
+import { ajax } from "discourse/lib/ajax";
+import { extractError } from "discourse/lib/ajax-error";
+import { bind } from "discourse-common/utils/decorators";
+import { i18n } from "discourse-i18n";
+
+const BULK_DELETE_CHANNEL = "/bulk-user-delete";
+
+export default class BulkUserDeleteConfirmation extends Component {
+  @service messageBus;
+
+  @tracked confirmButtonDisabled = true;
+  @tracked deleteStarted = false;
+  @tracked logs = new TrackedArray();
+  failedUsernames = [];
+
+  callAfterBulkDelete = false;
+
+  constructor() {
+    super(...arguments);
+    this.messageBus.subscribe(BULK_DELETE_CHANNEL, this.onDeleteProgress);
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.messageBus.unsubscribe(BULK_DELETE_CHANNEL, this.onDeleteProgress);
+  }
+
+  get confirmDeletePhrase() {
+    return i18n(
+      "admin.users.bulk_actions.delete.confirmation_modal.confirmation_phrase",
+      { count: this.args.model.userIds.length }
+    );
+  }
+
+  #logError(line) {
+    this.#log(line, "error");
+  }
+
+  #logSuccess(line) {
+    this.#log(line, "success");
+  }
+
+  #logNeutral(line) {
+    this.#log(line, "neutral");
+  }
+
+  #log(line, type) {
+    this.logs.push({
+      line,
+      type,
+    });
+  }
+
+  @bind
+  onDeleteProgress(data) {
+    if (data.success) {
+      this.#logSuccess(
+        i18n(
+          "admin.users.bulk_actions.delete.confirmation_modal.user_delete_succeeded",
+          {
+            position: data.position,
+            total: data.total,
+            username: data.username,
+          }
+        )
+      );
+    } else if (data.failed) {
+      this.failedUsernames.push(data.username);
+      this.#logError(
+        i18n(
+          "admin.users.bulk_actions.delete.confirmation_modal.user_delete_failed",
+          {
```

Comment:
Could just pass `data` as an argument to `i18n` here instead of specifying which keys, I don't think it would error